src/sas/qtgui/Perspectives/Inversion/InversionWidget.py

```diff
@@ -0,0 +1,1333 @@
+            GuiUtils.updateModelItemWithPlot(self._data, self.prPlot, title)
+            self.communicate.plotRequestedSignal.emit([self._data, self.prPlot], None)
+        if self.dataPlot is not None:
+            title = self.dataPlot.name
+            self.dataPlot.plot_role = Data1D.ROLE_DEFAULT
+            self.dataPlot.symbol = "Line"
+            self.dataPlot.show_errors = False
+            GuiUtils.updateModelItemWithPlot(self._data, self.dataPlot, title)
+            self.communicate.plotRequestedSignal.emit([self._data, self.dataPlot], None)
+        self.enableButtons()
+
+    def removeData(self, data_list=None):
+        """Remove the existing data reference from the P(r) Persepective"""
+        self.batchResults = {}
+        if self.is2D or self.isBatch:
+            self.prPlot = None
+            self.dataPlot = None
+            self.dataList.removeItem(self.dataList.currentIndex())
+            self._allowPlots = False
+        if not data_list:
+            data_list = [self._data]
+        self.closeDMax()
+        for data in data_list:
+            self._dataList.pop(data, None)
+        if self.dataPlot:
+            # Reset dataplot sliders
+            self.dataPlot.slider_low_q_input = []
+            self.dataPlot.slider_high_q_input = []
+            self.dataPlot.slider_low_q_setter = []
+            self.dataPlot.slider_high_q_setter = []
+        self._data = None
+        length = len(self.dataList)
+        for index in reversed(range(length)):
+            if self.dataList.itemData(index) in data_list:
+                self.dataList.removeItem(index)
+        # Last file removed
+        self.dataDeleted = False
+        if len(self._dataList) == 0:
+            self.prPlot = None
+            self.dataPlot = None
+            self.logic.data = None
+            self._calculator = Invertor()
+            self.closeBatchResults()
+            self.nTermsSuggested = NUMBER_OF_TERMS
+            self.noOfTermsSuggestionButton.setText("{:n}".format(
+                self.nTermsSuggested))
+            self.regConstantSuggestionButton.setText("{:-3.2g}".format(
+                REGULARIZATION))
+            self.updateGuiValues()
+            self.setupModel()
+        else:
+            self.dataList.setCurrentIndex(0)
+            self.updateGuiValues()
+
+    def getPage(self):
+        """
+        serializes full state of this fit page
+        """
+        # Get all parameters from page
+        param_dict = self.getState()
+        param_dict['data_name'] = str(self.logic.data.name)
+        param_dict['data_id'] = str(self.logic.data.id)
+        return param_dict
+
+    def currentTabDataId(self):
+        """
+        Returns the data ID of the current tab
+        """
+        tab_id = []
+        if self.logic.data_is_loaded:
+            tab_id.append(str(self.logic.data.id))
+        return tab_id
+
+    def updateFromParameters(self, params):
```

Comment:
Tab index will need to be added to the parameters dictionary so all tabs can be saved/loaded appropriately.

Reply:
Yes, Ill try to assign IDs for each of the tab I believe this should also fix [this](https://github.com/SasView/sasview/pull/2181#discussion_r966385534).

Reply:
Done